pkg/image/api/helper.go

```diff
@@ -191,6 +178,61 @@ func ParseDockerImageReference(spec string) (DockerImageReference, error) {
 	return ref, nil
 }
 
+// ParseDockerImageReference parses a Docker pull spec string into a
+// DockerImageReference.
+func ParseDockerImageReference(spec string) (DockerImageReference, error) {
+	// TODO replace with docker version once docker/docker PR11109 is merged upstream
+	stream, tag, id := parseRepositoryTag(spec)
+
+	ref, err := parseDockerImageReference(strings.Split(stream, "/"), spec)
+	if err != nil {
+		return ref, err
+	}
+	ref.Tag = tag
+	ref.ID = id
+
+	return ref, err
+}
+
+func ParseMultiSegmentsDockerImageReference(spec string) (DockerImageReference, error) {
+	stream, tag, id := parseRepositoryTag(spec)
+
+	repoParts := strings.Split(stream, "/")
+	if len(repoParts) < 4 {
+		ref, err := parseDockerImageReference(repoParts, spec)
+		if err != nil {
+			return ref, err
+		}
+		ref.Tag = tag
+		ref.ID = id
+
+		return ref, err
+	}
+
+	var ref DockerImageReference
+
+	// registry/namespace/name/namecomponent/namecomponent/...
+	if strings.HasSuffix(repoParts[0], ":") || IsRegistryDockerHub(repoParts[0]) {
+		return ref, fmt.Errorf("the docker pull spec %q must be two or three segments separated by slashes", spec)
```

Comment:
@miminar because this case for `http://example.com/namespace/name`. 

But I just realized that I can't act like that and in case of specifying of schema I need to parse the rest path. Thanks!